clrtrust.in

```diff
@@ -166,14 +182,20 @@ c_rehash_internal() {
     fi
     (
         cd "$dir"
-        find . -maxdepth 1 -type f | while read f; do
-            name=$(openssl x509 -in "$f" -noout -subject_hash 2>/dev/null)
+        hashes=$(find . -maxdepth 1 -type f | "${CLR_HELPER_CMD}" -s)
+        if [ $? -ne 0 ]; then
+            return 1
+        fi
+        echo "$hashes" | while IFS=$'\t' read f h; do
             lnno=0
-            while ! ln -s "$f" "${name}.${lnno}" && ((lnno++ < 100)); do
+            while ! ln -s "$f" "${h}.${lnno}" && ((lnno++ < 100)); do
                 :
             done 2>/dev/null
         done
     )
+    if [ $? -ne 0 ]; then
```

Comment:
Unless you want to explicitly want to return just 0/1, you could loose lines 196-199, as the function by default will return the value of the last command (in this case the subshell) by default.